### PR TITLE
test: 80% coverage of task component

### PR DESCRIPTION
**Pull request by suyashpatil78**
### <samp>🤖 Generated by Copilot at 09c632b</samp>

This pull request improves the test coverage for the dashboard tasks component. It adds unit tests for the `onTaskCardClick` and `onAdvanceRequestTaskCardClick` methods in `tasks-2.component.spec.ts`. It also uses mock data for testing the advance request task.

### <samp>🤖 Generated by Copilot at 09c632b</samp>

> _We're testing the `click` events on the dashboard_
> _We're mocking the data for the `advance request`_
> _Heave away, me hearties, heave away with zest_
> _We'll make the code shipshape and pass the test_

### <samp>🤖 Generated by Copilot at 09c632b</samp>

*  Import mock data for advance request testing ([link](https://github.com/fylein/fyle-mobile-app/pull/2376/files?diff=unified&w=0#diff-44e20b678ab2dba22c9272a91df9ed3bc9b18eff24367cf78355f20a38a32fa4R37))
*  Add tests for click handlers of task cards ([link](https://github.com/fylein/fyle-mobile-app/pull/2376/files?diff=unified&w=0#diff-44e20b678ab2dba22c9272a91df9ed3bc9b18eff24367cf78355f20a38a32fa4R297-R422))

## Clickup
https://app.clickup.com/t/85ztx3mbc

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/dashboard/tasks/tasks-2.component.spec.ts [hunk 1]
```diff
@@ -293,5 +294,131 @@ export function TestCases2(getTestBed) {
...
+        expect(loaderService.hideLoader).toHaveBeenCalledTimes(1);
+        expect(router.navigate).toHaveBeenCalledOnceWith([
+          '/',
+          'enterprise',
+          'add_edit_advance_request',
+          { id: singleExtendedAdvReqRes.data[0].areq_id },
+        ]);
+      }));
+
+      it('should navigate to my advances page if task count is greater than 1', fakeAsync(() => {
```
Comment (Dimple16) [inline src/app/fyle/dashboard/tasks/tasks-2.component.spec.ts:328]: the description says the task count is greater than 1, but we aren't setting the count value in this test case. Is this intentional?

File: src/app/fyle/dashboard/tasks/tasks-2.component.spec.ts [hunk 2]
```diff
@@ -293,5 +294,131 @@ export function TestCases2(getTestBed) {
...
+        expect(router.navigate).toHaveBeenCalledOnceWith([
+          '/',
+          'enterprise',
+          'view_team_report',
+          { id: apiReportRes.data[0].rp_id, navigate_back: true },
+        ]);
+      }));
+
+      it('should navigate to my reports page if task count is greater than 1', fakeAsync(() => {
+        component.onTeamReportsTaskClick(taskCtaData3, dashboardTasksData[0]);
```
Comment (Dimple16) [inline src/app/fyle/dashboard/tasks/tasks-2.component.spec.ts:372]: same as above

**Review by Dimple16 — CHANGES_REQUESTED** [review]
few questions

File: src/app/fyle/dashboard/tasks/tasks-2.component.spec.ts (hunk 1 above)
Comment (suyashpatil78) [reply src/app/fyle/dashboard/tasks/tasks-2.component.spec.ts:328]: @Dimple16  We are not setting count value as it is already done in mock data
Screenshot 2023-09-07 at 6 25 11 PM

File: src/app/fyle/dashboard/tasks/tasks-2.component.spec.ts (hunk 2 above)
Comment (suyashpatil78) [reply src/app/fyle/dashboard/tasks/tasks-2.component.spec.ts:372]: ref - https://github.com/fylein/fyle-mobile-app/pull/2376#discussion_r1318563056

**Review by Dimple16 — APPROVED** [review]

File: src/app/fyle/dashboard/tasks/tasks-2.component.spec.ts [hunk 3]
```diff
@@ -293,5 +294,131 @@ export function TestCases2(getTestBed) {
...
+      }));
+
+      it('should navigate to my advances page if task count is greater than 1', fakeAsync(() => {
+        component.onSentBackAdvanceTaskClick(taskCtaData3, dashboardTasksData[0]);
+        tick(100);
+        expect(loaderService.showLoader).not.toHaveBeenCalled();
+        expect(advanceRequestService.getMyadvanceRequests).not.toHaveBeenCalled();
+        expect(loaderService.hideLoader).not.toHaveBeenCalled();
+        expect(router.navigate).toHaveBeenCalledOnceWith(['/', 'enterprise', 'my_advances'], {
+          queryParams: { filters: '{"state":["SENT_BACK"]}' },
```
Comment (OmkarJ13) [inline src/app/fyle/dashboard/tasks/tasks-2.component.spec.ts:335]: Can we use `JSON.stringify` for cases like this?

**Review by OmkarJ13 — APPROVED** [review]

File: src/app/fyle/dashboard/tasks/tasks-2.component.spec.ts (hunk 3 above)
Comment (suyashpatil78) [reply src/app/fyle/dashboard/tasks/tasks-2.component.spec.ts:335]: yes, we can use that

Comment (suyashpatil78) [reply src/app/fyle/dashboard/tasks/tasks-2.component.spec.ts]: done

**Pull request merged**